src/Bridge/Doctrine/ClassMetadata/PropertyHelper.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\ClassMetadata;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\Common\Persistence\Mapping\ClassMetadata;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * Helper for getting information regarding a property using the resource metadata.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class PropertyHelper
```

Comment:
As it's a final class, we must also introduce an interface.
Also, can you move it in the existing `Common` namespace, it doesn't worth introducing a new namespace for a single class.

Reply:
Another solution, that solves both the DI and the interface problem, is to transform this class in a trait.

Reply:
OK for the interface and for moving it.
I don't really like the idea of having a trait though. I think this service makes sense. It exposes methods to know information about a property using the Doctrine class metadata.

Reply:
Its something purely internal tight to Doctrine. Adding this service make our public API broader (so harder to maintain), and increase the overall complexity for no real benefit (ex: in the compiler pass).
It also makes it harder to use API Platform without a framework, or other frameworks than SF (more class to initialize).

To me a trait is enough, and will make the PR smaller. WDYT @api-platform/core-team?

Reply:
I'm :+1: for the trait